bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/airqHandler.java

```diff
@@ -0,0 +1,936 @@
+                requestMethod, body);
+        try {
+            conn = (HttpURLConnection) new URL(address).openConnection();
+            conn.setRequestMethod(requestMethod);
+            conn.setRequestProperty("Content-Type", "application/json");
+            conn.setConnectTimeout(timeout);
+            conn.setReadTimeout(timeout);
+            if (body != null && !body.isEmpty()) {
+                conn.setDoOutput(true);
+                try (Writer out = new OutputStreamWriter(conn.getOutputStream())) {
+                    out.write(body);
+                }
+            }
+            try (InputStream in = conn.getInputStream(); ByteArrayOutputStream result = new ByteArrayOutputStream()) {
+                byte[] buffer = new byte[1024];
+                int length;
+                while ((length = in.read(buffer)) != -1) {
+                    result.write(buffer, 0, length);
+                }
+                conn.disconnect();
+                return new Result(result.toString(StandardCharsets.UTF_8.name()), conn.getResponseCode());
+            }
+        } catch (IOException exc) {
+            if (System.out != null) {
+
+                {
+                    System.out.println(
+                            "air-Q - airqHandler - doNetwork(): Error while accessing air-Q: " + exc.toString());
+                }
+            }
+        } finally {
+            if (conn != null) {
+                conn.disconnect();
+            }
+        }
+        return null;
+    }
+
+    public static class Result {
+        private final String body;
+        private final int responseCode;
+
+        public Result(String body, int responseCode) {
+            this.body = body;
+            this.responseCode = responseCode;
+        }
+
+        public String getBody() {
+            return body;
+        }
+
+        public int getResponseCode() {
+            return responseCode;
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        if (getConfigDataJob != null) {
+            getConfigDataJob.cancel(true);
+        }
+    }
+
+    public void pollData() {
+        logger.trace("air-Q - airqHandler - run(): starting polled data handler");
+        if ((!ipaddress.isEmpty()) && (!password.isEmpty())) {
+            try {
+                String url = "http://".concat(ipaddress.concat("/data"));
+                String jsonAnswer = getDecryptedContentString(url, "GET", null);
+                if (jsonAnswer != null) {
+                    Gson gson = new Gson();
+                    JsonElement decEl = gson.fromJson(jsonAnswer, JsonElement.class);
+                    if (decEl != null) {
+                        JsonObject decObj = decEl.getAsJsonObject();
+                        logger.debug("air-Q - airqHandler - run(): decObj={}", decObj);
+                        // processType(decObj, "bat", "battery", "pair");
```

Comment:
remove commented code.

Reply:
I added an explanation about why 'bat' is commented out. I have the same in thing-types.xml. The code ready to be used, but not applicable at this moment because no air-Q device is battery powered yet.

Reply:
I'll leave this conversation open for the next reviewer to look at.